src/scripts.js

```diff
@@ -172,39 +148,224 @@ function showInfo() {
+// window.addEventListener("load", fetchData);
+
+function fetchData() {
+  const userInfo = fetchCalls.callFitLitData('users');
+  const activityInfo = fetchCalls.callFitLitData('activity');
+  const hydrationInfo = fetchCalls.callFitLitData('hydration');
+  const sleepInfo = fetchCalls.callFitLitData('sleep');
+
+  Promise.all([userInfo, activityInfo, hydrationInfo, sleepInfo])
+  .then(data => initializedData(data[0], data[1], data[2], data[3]))
+  .catch(err => console.error(err))
 }
 
-function updateTrendingStepDays() {
-  user.findTrendingStepDays();
-  trendingStepsPhraseContainer.innerHTML = `<p class='trend-line'>${user.trendingStepDays[0]}</p>`;
+function initializedData(userData, activityData, hydrationData, sleepData) {
+  Promise.resolve(intializeUserData(userData)).then(storeUserData(activityData, hydrationData, sleepData)).then(updatePageInfo());
 }
 
-for (var i = 0; i < dailyOz.length; i++) {
-  dailyOz[i].innerText = user.addDailyOunces(Object.keys(sortedHydrationDataByDate[i])[0])
+function intializeUserData(userData) {
+  userData.userData.forEach(user => {
+    let userInstance = new User(user);
+    userRepository.users.push(userInstance)
+  })
 }
 
-dropdownGoal.innerText = `DAILY STEP GOAL | ${user.dailyStepGoal}`;
+fetchData();
+////////////////// FETCH CALLS -------------------------------->
+
+//This will then take the data from the api call data and AFTER the use repo is created with all of the user instances it can then store the user data on the user.
+function storeUserData (activityData, hydrationData, sleepData) {
+    // -------------------------------------->
+    // After that we had instantiated every element from the userData in a User class, we would update the properties relted of each instated user (.activityRecord, .accomplishedDays, .trendingStepDays ...) - Using this iteration will allow us to create an instances of every element from the activityData file and push it in the correct instantiated user ! ---->
+    
+    activityData.activityData.forEach(activity => {
+     new Activity(activity, userRepository);
+    });
+    //--------------------------------------->
+
+    // The same idea for this other two data sets ---------------->
+    hydrationData.hydrationData.forEach(hydration => {
+      new Hydration(hydration, userRepository);
+    });
+
+    sleepData.sleepData.forEach(sleep => {
+      new Sleep(sleep, userRepository);
+    });
+    // ----------------------------------------------------------->
+  }
 
-dropdownEmail.innerText = `EMAIL | ${user.email}`;
 
-dropdownName.innerText = user.name.toUpperCase();
+  function updatePageInfo() {
+    let user = userRepository.users[0];
+    
+    activityInformation(user, userRepository);
+    sleepInformation(user, userRepository);
+    userInformation(user);
+    //NEED to FIX hydration function here// this is why it currently does not show up on the page.. there is no function that calls it.
```

Comment:
The hydration data is not displayed correctly. See error message in the console.